src/main/java/com/contaazul/dsl/DateBuilder.java

```diff
@@ -35,28 +35,31 @@ public DateBuilder(Date date) {
 	/**
 	 * Ajout de TimeUnit a une date.
 	 * 
-	 * @param unit
+	 * @param timeUnit
 	 *            TimeUnit
 	 * @return DateBuilder
 	 */
-	public DateBuilder add(TimeUnit unit) {
-
-		if (unit.workingDay) {
-			int nbDay = unit.size;
-			int i = 0;
-			while (i != nbDay) {
-
-				date.add( unit.type, 1 );
+	public DateBuilder add(TimeUnit timeUnit) {
+		if (timeUnit.workingDay)
+			moveWorkingDays( timeUnit );
+		else
+			date.add( timeUnit.type, timeUnit.size );
+		return this;
+	}
 
-				if (!isWeekend()) {
-					i++;
-				}
-			}
+	private void moveWorkingDays(TimeUnit timeUnit) {
+		int moves = timeUnit.size;
+		int moved = 0;
+		int nextMove = (moves > 0) ? 1 : -1;
+		while (moved != moves)
+			moved = moveWorkingDay( timeUnit, moved, nextMove );
+	}
 
-		} else {
-			date.add( unit.type, unit.size );
-		}
-		return this;
+	private int moveWorkingDay(TimeUnit unit, int moved, int move) {
+		date.add( unit.type, move );
+		if (!isWeekend())
```

Comment:
É isso mesmo, este método adiciona "dias úteis", se for final de semana não conta no total de "dias úteis" que se quer movimentar.

Se hoje é sexta-feira e eu quero avançar 1 dia útil, então sábado e domingo não podem contar.